### ISSUE TITLE
Stop bypassing Pkg resolution for version numbers

### ISSUE THREAD

**Issue by Keno** [JuliaPackaging/BinaryBuilderBase.jl#328]
The version selection here:

https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/18e421fac6cf7f8e3c682500033d98934ea086ee/src/Dependencies.jl#L117-L122

is problematic, because it doesn't consider whether the minimum selected version is feasible. Instead, the full compat should be passed to Pkg's version resolution (if possible with a request to resolve the minimum feasible version).

This came up in https://github.com/JuliaPackaging/Yggdrasil/pull/7411 which has complicated dependencies and it turns out the minimum compatible MPI version was incompatible with other dependencies.

**Comment by giordano** [comment]
You can pass the exact version number to use for the build as second argument to [`Dependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.Dependency) if that helps.

**Comment by Keno** [comment]
Yes, I fixed it that way, but this compat was selected by `platforms/mpi.jl`, so it was a bit of a pain.
